feature-gate/program/src/processor.rs

```diff
@@ -7,12 +7,41 @@ use {
         entrypoint::ProgramResult,
         feature::Feature,
         msg,
+        program::invoke,
         program_error::ProgramError,
         pubkey::Pubkey,
-        system_program,
+        system_instruction, system_program,
     },
 };
 
+/// Processes an [ActivateFeature](enum.FeatureGateInstruction.html)
+/// instruction.
+pub fn process_activate_feature(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+
+    let feature_info = next_account_info(account_info_iter)?;
+    let _system_program_info = next_account_info(account_info_iter)?;
+
+    if !feature_info.is_signer {
+        return Err(ProgramError::MissingRequiredSignature);
+    }
+
+    if feature_info.owner != &system_program::id() {
+        return Err(FeatureGateError::InvalidFeatureAccount.into());
+    }
+
```

Comment:
I'm advocating for moving the transfer into this processor, but if we don't we definitely need to check the lamport balance here.